### PR TITLE
fix: 🍰 Checkboxes Not Missing Anymore On Delete User Account Page

### LINKED ISSUE #3498: 🐛 [Bug] Checkboxes missing on delete user account page
## :bug: Bugreport

The checkboxes on the delete user account page are missing in the code of the frontend. Currently, users cannot delete their posts and comments when deleting their account.

hc-delete-user-account-checkboxes-missing

### Steps to reproduce the behavior
1. Log into the network.
2. Go to the delete user account page.
3. Checkboxes missing

### Expected behavior

The checkboxes should be available to enable users to delete their posts and comments.

### Version & Environment
 Type: [Notebook]      
 - OS: [macOS]      
 - Browser: [Firefox] 
 - Version [75.0]  
 - Device: [MacBook Pro]  

### Additional context

Checked with @mattwr18

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
the checkboxes for saved contributions and comments of a user are displayed again at userDelete

### Issues
- fixes #3498

### CODE REVIEW

**Comment by rbeer** [comment]
Am I correct in assuming that the problem was that the length were missing, so that these `v-if` never come true?

https://github.com/Human-Connection/Human-Connection/blob/9aacb22a582ff72c90c1c46c2480a506c9f5d6a8/webapp/components/DeleteData/DeleteData.vue#L12

**Comment by ogerly** [comment]
I have removed the usercount from the currentUser. 
the usercounts are no longer queried on the deleteUser. the checkboxes are displayed as cecked by default. 

FireShot Capture 037 - Human Connection - Human Connection - localhost

FireShot Capture 036 - Human Connection - Human Connection - localhost

**Comment by ogerly** [comment]
deleteUser auf altem stand, currentUserCounts werden zusätzlich abgerufen, ich habe dazu un der 
 webapp/graphql/User.js eine currentUserCountQuery() angelegt. 
allerdings ist das irgendwie doppelt gemoppelt. 
aber es ist nun nur in der delete-data page vorhanden. 

ich habe aber den admin ausgeschlossen, einfach das schiff zu verlassen. hier müsste vorher ein superadmin her. da der nicht da ist. darf ein admin sich über usersettings nicht löschen lassen können dürfen sollen. 
 
 und ich habe für die i18n in vuex die zählvariante gefunden und sie geht gut .. ;) 


FireShot Capture 048 - vuex-i18n - npm - www npmjs com

**Comment by ogerly** [comment]
sorry

File: webapp/components/DeleteData/DeleteData.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <base-card class="delete-data">
+  <base-card v-if="currentUser.role !== 'admin'" class="delete-data">
```
Comment (mattwr18) [inline webapp/components/DeleteData/DeleteData.vue:2]: I don't think we have agreed on this, therefore I would open an issue and we can discuss it, not just make this decision alone.

Comment (ogerly) [reply webapp/components/DeleteData/DeleteData.vue]: so i changed it

File: webapp/components/DeleteData/DeleteData.vue [hunk 2]
```diff
@@ -52,8 +61,19 @@ export default {
       deleteContributions: false,
       deleteComments: false,
       enableDeletionValue: null,
+      currentUserCount: [],
```
Comment (Tirokk) [inline webapp/components/DeleteData/DeleteData.vue:64]: This makes sense here to have this in a separate variable in my eyes.

I would name it with plural `currentUserCounts`.
And this is initial no array. It’s an object:
`currentUserCounts: {},`

File: webapp/graphql/User.js [hunk 3]
```diff
@@ -265,9 +265,11 @@ export const checkSlugAvailableQuery = gql`
 
 export const currentUserQuery = gql`
   ${userFragment}
+  ${userCountsFragment}
```
Comment (Tirokk) [inline webapp/graphql/User.js:268]: I think you have forgotten to delete this?

File: webapp/graphql/User.js [hunk 4]
```diff
@@ -265,9 +265,11 @@ export const checkSlugAvailableQuery = gql`
...
   query {
     currentUser {
       ...user
+      ...userCounts
```
Comment (Tirokk) [inline webapp/graphql/User.js:272]: I think you have forgotten to delete this?

File: webapp/graphql/User.js [hunk 5]
```diff
@@ -283,3 +285,12 @@ export const currentUserQuery = gql`
     }
   }
 `
+
+export const currentUserCountQuery = () => gql`
```
Comment (Tirokk) [inline webapp/graphql/User.js:289]: Great you did a separate query !!! 😍

**Comment by Tirokk** [comment]
I have fixed the listing …

**Pull request merged**